utility/improve-punctuation.js

```diff
@@ -0,0 +1,19 @@
+module.exports = improvePunctuation
+
+var LEFT_DOUBLE = '“'
+var RIGHT_DOUBLE = '”'
+var RIGHT_SINGLE = '’'
+
+var replacements = [
+  [ /^"/g, LEFT_DOUBLE ],
+  [ /"$/g, RIGHT_DOUBLE ],
+  [ / "/g, ( ' ' + LEFT_DOUBLE ) ],
+  [ /"/g,  RIGHT_DOUBLE ],
+  [ /'/g,  RIGHT_SINGLE] ]
```

Comment:
These substitutions are applied in order. That order reflects a certain amount of "playing the odds". In effect, double quotes are presumed closing/right double quotes, as at the end of a quotation, and every single quote is presumed as right/closing quote, as in a contraction.